python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -145,6 +162,31 @@ def add_node(node_type, available_resources=None):
 
         return node_resources, node_type_counts
 
+    def reserve_and_allocate_spread(self,
+                                    strict_spreads: List[List[ResourceDict]],
+                                    node_resources: List[ResourceDict],
+                                    node_type_counts: Dict[NodeType, int]):
```

Comment:
Can you please add type annotation to the return value and provide a description of the function with Args/Returns?